pymc_experimental/model/marginal_model.py

```diff
@@ -208,35 +210,50 @@ def logp(self, vars=None, **kwargs):
             vars = [m[var.name] for var in vars]
         return m._logp(vars=vars, **kwargs)
 
-    def clone(self):
-        m = MarginalModel(coords=self.coords)
-        model_vars = self.basic_RVs + self.potentials + self.deterministics + self.marginalized_rvs
-        data_vars = [var for name, var in self.named_vars.items() if var not in model_vars]
+    @staticmethod
+    def _clone(model: Union[Model, "MarginalModel"]):
```

Comment:
This is used internally to clone itself, I just made it a static method so it can be used outside of instances.

It's a staticmethod because it doesn't access any cls attribute. I made it so I can use it from the new function

Reply:
But yes `from_model` may be better. It just happens to `clone` as a side-effect of creating the new model. Will rename

Reply:
It's not weird to `return MarginalModel(...)` from inside `MarginalModel` instead of `return cls(...)` ?

Reply:
Why? It's just a function that exists inside the MarginalModel class. It forbids subclassing, which is probably a good thing. Even subclassing the base Model is tricky as is